research-software-directory/RSD-as-a-service
# 247 project and software maintainers

# Initial maintainer invitation flow for projects and software

Changes proposed in this pull request:

* Add an invitation table in the database for projects
* Add a function to the database to accept an invitation to become a maintainer for a project
* Add a function to the database that allows you to see information of the other maintainers of a project that you maintain
* The frontend facilitates using these features by allowing to generate invitation urls, by giving an overview of maintainers and by allowing to delete maitnainers

How to test:

* Rebuild the project
	* `docker-compose down --volumes && docker-compose build --parallel && docker-compose up --scale scrapers=0`
* Login and create a project
* On the admin page, go to maintainers
* Generate an invite url
* Log in as a different user and go to the invite url you generated above
* You should now be a maintainer of that project
* Try to delete a maintainer, when one or on the last maintainer delete icon is disabled

Closes #247
Closes #258

PR Checklist:
